### DISCUSSION TITLE
Creating thermodynamic databases

### DISCUSSION THREAD

**Discussion by JMCarmona10** [reaktoro/reaktoro#361] [Q&A]
### Description

Hello, Allan 

I have a question about measuring how the initial concentration of a species changes compared to its equilibrium concentration after a reaction finishes. I created a thermodynamic database with two phases - aqueous and solid - and included 8 reactions, none involving gases. However, when I try to determine how the initial concentrations change over the course of the reaction to reach equilibrium, the concentrations remain unchanged. I realized that with the reactions I defined, I can only calculate equilibrium concentrations for the products formed, not observe how the concentrations of all species change over time to reach equilibrium.  

### What is your operating system?

None

**Comment by allanleal** [comment]
Hi Manuel, thanks for your question. It’s very difficult to know what is going on without a reproducible example (one that is simple and to the point). Would you be able to provide it?

**Reply by JMCarmona10** [reply]
Please accept my apologies for the late reply. I was able to solve the initial issue on my own. However, I have another question related to creating thermodynamic databases. Recently, I discovered that when expressing an oil  phase  in a .yaml file to run in Reaktoro, I can perform chemical equilibrium calculations without specifying thermodynamic properties like G, H, and Cp. The simulation runs with only the chemical reactions and equilibrium constants defined. Would eliminating these properties have any impact on the results?

**Reply by allanleal** [reply]
Hi @JMCarmona10, there are many ways to specify these standard thermodynamic properties and models in a Reaktoro database. It's hard to answer without seeing exactly what you're doing, but for conventional equilibrium calculations with T and P given, you don't need H0, Cp0. You should need information to calculate G0(T, P) unless you are using logKs of reactions.

**Reply by JMCarmona10** [reply]
I am writing with a question about creating thermodynamic databases. Specifically, I would like to know if there is a way to fix the equilibrium constants for a reaction at different temperatures within the software.

I have calculated equilibrium values for my reaction across a temperature range. However, I want to understand how valid these values are based on the models and data in the software.

**Reply by allanleal** [reply]
Do you mean providing logK data for interpolation? This is currently not supported. Only standard Gibbs energies and other standard-state properties can be provided for interpolation (as shown in the link below for species Ca+2). 

https://reaktoro.org/tutorials/advanced/creating-custom-databases.html?highlight=interpolation#a-more-complicated-custom-thermodynamic-database-in-yaml-format

**Reply by JMCarmona10** [reply]
Rather than interpolating logK values, im tryingto use experimental values at different temperatures. For example, the following two reactions had logK values that varied with temperature in the experiment.  Im trying to create a database close to the experiment environment as close as possible. And then run the chemical equilibrium.  

![Screenshot from 2024-02-20 16-14-15](https://github.com/reaktoro/reaktoro/assets/116467897/ec3782d6-6a2b-4bfc-8bb1-57ca28cc453f)

I came across some code that assigns logK values to different chemical reactions. I'm wondering if it would be possible to modify the code to specify logK values for individual reactions at different temperatures.  
![Screenshot from 2024-02-20 16-20-31](https://github.com/reaktoro/reaktoro/assets/116467897/e2137989-5fe0-4642-83f4-a1974b532aa7)

**Reply by allanleal** [reply]
OK, try the following. First, you define a function `model` (any name possible) that returns a `ReactionStandardThermoProps` object (consult https://reaktoro.org/api).

```py
def model(args):
    T = args.T  # temperature in K
    P = args.P  # pressure in Pa
    logK = 1.23 + 2.34/T - 3.45e-3*T  # compute log(K) for current T
    lnK = logK * math.log(10)  # convert log(K) to ln(K)
    
    rprops = rkt.ReactionStandardThermoProps()
    rprops.dG0 = -R*T*lnK  # compute dG0 for the reaction using lnK

    return rprops
```

Then, maybe, you may have to convert it to `ReactionStandardThermoModel` as follows:

```py
model = rkt.ReactionStandardThermoModel(model)  # create a reaction model
```

Now, instead of using `.withEquilibriumConstant` in the screenshot, you use `.withReactionStandardThermoModel(model)`.

You can do this for other reactions as well. 

If you run into issues, please consider creating a very minimal script reproducing the problem and attach here.

**Reply by JMCarmona10** [reply]
Thank you i was able to write all my reactions and now it matches the chemical system of the experiment
